components/core/tests/test-StreamingCompression.cpp

```diff
@@ -1,230 +1,102 @@
-                == decompressor.get_decompressed_stream_region(
-                        uncompressed_bytes,
-                        decompressed_data,
-                        uncompressed_data_size / 5
-                ));
-        REQUIRE(memcmp(uncompressed_data, decompressed_data, uncompressed_data_size / 5) == 0);
-        memset(decompressed_data, 0, uncompressed_data_size);
-        uncompressed_bytes += uncompressed_data_size / 5;
-        REQUIRE(ErrorCode_Success
-                == decompressor.get_decompressed_stream_region(
-                        uncompressed_bytes,
-                        decompressed_data,
-                        uncompressed_data_size / 2
-                ));
-        REQUIRE(memcmp(uncompressed_data, decompressed_data, uncompressed_data_size / 2) == 0);
-        memset(decompressed_data, 0, uncompressed_data_size);
-        uncompressed_bytes += uncompressed_data_size / 2;
-        REQUIRE(ErrorCode_Success
-                == decompressor.get_decompressed_stream_region(
-                        uncompressed_bytes,
-                        decompressed_data,
-                        uncompressed_data_size
-                ));
-        REQUIRE(memcmp(uncompressed_data, decompressed_data, uncompressed_data_size) == 0);
-        memset(decompressed_data, 0, uncompressed_data_size);
-        uncompressed_bytes += uncompressed_data_size;
+    Array<char> decompressed_buffer{cBufferSize};
 
-        // Cleanup
-        boost::filesystem::remove(compressed_file_path);
+    // Compress
+    FileWriter file_writer;
+    file_writer.open(compressed_file_path, FileWriter::OpenMode::CREATE_FOR_WRITING);
+    compressor->open(file_writer);
+    for (auto const chunk_size : cCompressionChunkSizes) {
+        compressor->write(uncompressed_buffer.data(), chunk_size);
+    }
+    compressor->close();
+    file_writer.close();
+
+    // Decompress and compare
+    clp::ReadOnlyMemoryMappedFile const memory_mapped_compressed_file{compressed_file_path};
+    auto const compressed_file_view{memory_mapped_compressed_file.get_view()};
+    decompressor->open(compressed_file_view.data(), compressed_file_view.size());
+
+    size_t num_uncompressed_bytes{0};
+    for (auto const chunk_size : cCompressionChunkSizes) {
+        // Clear the buffer to ensure that we are not comparing values from a previous test
+        std::ranges::fill(decompressed_buffer.begin(), decompressed_buffer.end(), 0);
+        REQUIRE(
+                (ErrorCode_Success
+                 == decompressor->get_decompressed_stream_region(
+                         num_uncompressed_bytes,
+                         decompressed_buffer.data(),
+                         chunk_size
+                 ))
+        );
+        REQUIRE(std::equal(
+                uncompressed_buffer.begin(),
+                uncompressed_buffer.begin() + chunk_size,
+                decompressed_buffer.begin()
+        ));
+        num_uncompressed_bytes += chunk_size;
     }
```

Comment:
How about adding this in the end (as explained in the last round of review):
```cpp
    REQUIRE(
            (std::accumulate(cCompressionChunkSizes.cbegin(), cCompressionChunkSizes.cend(), 0)
             == num_uncompressed_bytes)
    );
```
<numeric>(https://en.cppreference.com/w/cpp/header/numeric) will be required